pkg/jx/cmd/shell.go

```diff
@@ -130,27 +131,23 @@ func (o *ShellOptions) Run() error {
 	newConfig := *config
 	newConfig.CurrentContext = ctxName
 
-	//clean old folders
-	files, err := filepath.Glob("/tmp/.jx-shell-*")
-	if err != nil {
-		panic(err)
-	}
-	for _, f := range files {
-		if err := os.RemoveAll(f); err != nil {
-			panic(err)
-		}
-	}
-	tmpDirName, err := ioutil.TempDir("/tmp", ".jx-shell-")
+	tmpDirName, err := ioutil.TempDir("", ".jx-shell-")
 	if err != nil {
 		return err
 	}
-	tmpConfigFileName := tmpDirName + "/config"
+	tmpConfigFileName := filepath.Join(tmpDirName,"/config")
 	err = clientcmd.WriteToFile(newConfig, tmpConfigFileName)
 	if err != nil {
 		return err
 	}
 
-	shell := filepath.Base(os.Getenv("SHELL"))
+	fullShell := os.Getenv("SHELL")
+	shell := filepath.Base(fullShell)
+	if fullShell == "" && runtime.GOOS == "windows" {
+		// SHELL is set by git-bash but not cygwin :-(
```

Comment:
this should not make things any worse that they are for cygwin users as before it would try to exec `.` which would fail.